planner/cascades/enforcer_rules.go

```diff
@@ -33,15 +35,23 @@ type Enforcer interface {
 
 // GetEnforcerRules gets all candidate enforcer rules based
 // on required physical property.
-func GetEnforcerRules(prop *property.PhysicalProperty) (enforcers []Enforcer) {
+func GetEnforcerRules(g *memo.Group, prop *property.PhysicalProperty) (enforcers []Enforcer) {
+	if g.EngineType != memo.EngineTiDB {
+		return
+	}
 	if !prop.IsEmpty() {
+		orderEnforcer.group = g
```

Comment:
`orderEnforcer` is a global variable now, so we cannot write it concurrently? I don't get the point why we add those fields into `orderEnforcer`.

Reply:
Because I thought it looks terrible to create a PhysicalSort to calculate the cost of sort. I was trying to reuse the PhysicalSort struct.